util/chplenv/chpl_gpu.py

```diff
@@ -5,29 +5,52 @@
 import chpl_llvm
 from utils import error, memoize, run_command
 
+# Format:
+#   environment variable
+#   program to locate SDK folder
+#   depth of program within SDK folder
+GPU_TYPES = {
+    "cuda": ("CHPL_CUDA_PATH", "nvcc", 2),
+    "amd": ("CHPL_ROCM_PATH", "hipcc", 3)
+}
+
 def get():
-    if chpl_locale_model.get() == 'gpu':
-        return 'cuda'
-    else:
+    if chpl_locale_model.get() != 'gpu':
         return 'none'
 
-@memoize
-def get_cuda_path():
-    if chpl_locale_model.get() == 'gpu':
-        chpl_cuda_path = os.environ.get("CHPL_CUDA_PATH")
-        if chpl_cuda_path:
-            return chpl_cuda_path
-
-        exists, returncode, my_stdout, my_stderr = utils.try_run_command(["which",
-                                                                          "nvcc"])
-
-        if exists and returncode == 0:
-            chpl_cuda_path = "/".join(os.path.realpath(my_stdout).strip().split("/")[:-2])
-            return chpl_cuda_path
+    chpl_gpu_codegen = os.environ.get("CHPL_GPU_CODEGEN")
+    if chpl_gpu_codegen:
+        if chpl_gpu_codegen not in GPU_TYPES:
+            error("Only {} supported for 'CHPL_GPU_CODEGEN'".format(GPU_TYPES.keys()))
         else:
-            error("Can't find cuda")
+            return chpl_gpu_codegen
+    else:
+        return 'cuda'
 
-    return ""
+@memoize
+def get_sdk_path():
+    gpu_type = get()
+
+    # No SDK path if GPU is not being used.
+    if gpu_type == 'none':
+        return ''
+
+    # Check vendor-specific environment variable for SDK path, or
+    # try to find the SDK by running `which` on a vendor-specific program.
+    gpu_variable, gpu_program, gpu_bin_depth = GPU_TYPES[gpu_type]
+    chpl_sdk_path = os.environ.get(gpu_variable)
+    if chpl_sdk_path:
+        return chpl_sdk_path
+
+    exists, returncode, my_stdout, my_stderr = utils.try_run_command(["which",
```

Comment:
I realize this predates your PR but the use of the "which" command basically means this will only work on *nix systems (e.g. not Windows).

I guess this is acceptable for now though I should probably update the tech report to indicate this limitation.